GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -719,13 +719,13 @@ public IReadOnlyList<GitRevision> GetSelectedRevisions(SortDirection? direction
                 .ToList();
         }
 
-        private (ObjectId? firstId, GitRevision? selectedRev) getFirstAndSelected()
+        private (GitRevision? firstRev, GitRevision? selectedRev) GetFirstAndSelected()
         {
-            var revisions = GetSelectedRevisions();
-            var selectedRev = revisions?.FirstOrDefault();
-            var firstId = revisions is not null && revisions.Count > 1 ? revisions.LastOrDefault().ObjectId : selectedRev?.FirstParentId;
+            IReadOnlyList<GitRevision> revisions = GetSelectedRevisions();
 
-            return (firstId, selectedRev);
+            return revisions.Count > 0
+                ? (firstRev: revisions[revisions.Count - 1], selectedRev: revisions[0])
```

Comment:
❓

For `revisions.Count == 1` it returned `(selectedRev.FirstParent, selectedRev)`, but now `(selectedRev, selectedRev)`

❔
```suggestion
                ? (firstRev: revisions[^1], selectedRev: revisions[0])
```

Reply:
Changing the logic back for count==1
Not happy that GetRevision() is always called for count==1, may revert to

```
private (ObjectId? firstRev, GitRevision? selectedRev) GetFirstAndSelected()
```

Applying the change - should maybe be a separate PR to change to _index from end_ operator